account_payment_order_transfer_journal/tests/test_payment_order.py

```diff
@@ -0,0 +1,131 @@
+            {
+                "name": "Test Partner",
+            }
+        )
+        self.product = self.env["product.product"].create({"name": "Test"})
+        self.invoice_line_account = self.env["account.account"].create(
+            {
+                "name": "Test account",
+                "code": "TEST1",
+                "user_type_id": self.env.ref("account.data_account_type_expenses").id,
+            }
+        )
+        self.bank_journal = (
+            self.env["account.journal"].search(
+                [("company_id", "=", self.company.id), ("type", "=", "bank")], limit=1
+            ),
+        )[0]
+        self.transfer_journal = (
+            self.env["account.journal"].search(
+                [("company_id", "=", self.company.id), ("type", "=", "general")],
+                limit=1,
+            ),
+        )[0]
+
+        self.mode = self.env["account.payment.mode"].create(
+            {
+                "name": "Test Credit Transfer to Suppliers",
+                "company_id": self.company.id,
+                "payment_method_id": self.env.ref(
+                    "account.account_payment_method_manual_out"
+                ).id,
+                "fixed_journal_id": self.bank_journal.id,
+                "bank_account_link": "fixed",
+            }
+        )
+        self.invoice = self._create_supplier_invoice("F1242")
+        # Make sure no other payment orders are in the DB
+        self.domain = [
+            ("state", "=", "draft"),
+            ("payment_type", "=", "outbound"),
+            ("company_id", "=", self.env.user.company_id.id),
+        ]
+        self.env["account.payment.order"].search(self.domain).unlink()
+
+    def _create_supplier_invoice(self, ref):
+        invoice = self.env["account.move"].create(
+            {
+                "partner_id": self.partner.id,
+                "move_type": "in_invoice",
+                "ref": ref,
+                "payment_mode_id": self.mode.id,
+                "invoice_date": fields.Date.today(),
+                "invoice_line_ids": [
+                    (
+                        0,
+                        None,
+                        {
+                            "product_id": self.product.id,
+                            "quantity": 1.0,
+                            "price_unit": 100.0,
+                            "name": "product that cost 100",
+                            "account_id": self.invoice_line_account.id,
+                        },
+                    )
+                ],
+            }
+        )
+
+        return invoice
+
+    def test_payment_order_transfer_jounral(self):
+        self.invoice.action_post()
+        self.bank_journal.transfer_journal_id = self.transfer_journal
+        order_vals = {
+            "payment_type": "outbound",
+            "payment_mode_id": self.mode.id,
+        }
+        order = self.env["account.payment.order"].create(order_vals)
+        with self.assertRaises(UserError):
```

Comment:
You should check if the exception raised is the expected one
```
with self.assertRaises(UserError) as em:
    order.draft2open()
self.assertIn("A fragment message", em.exception.args[0])
```

Reply:
Why did you finally remove these 2 lines? Not relevant?

Reply:
Indeed. Already tested in account_payment_order